### ISSUE TITLE
Use Kata Containers rust crate for container ID verification

### ISSUE THREAD

**Issue by jodh-intel** [confidential-containers/enclave-cc#15]
## Background

As mentioned in https://github.com/confidential-containers/enclave-cc/pull/14#discussion_r913100448, that PR includes some code from the Kata Containers [agent](https://github.com/kata-containers/kata-containers/tree/main/src/agent) (added on https://github.com/kata-containers/kata-containers/pull/1521).

## Upcoming Kata changes

Hence, there are now two versions of this code. However, the original version has been moved into a separate rust crate in the Kata 3.x `runtime-rs` branch:

- https://github.com/kata-containers/kata-containers/blob/runtime-rs/src/libs/kata-sys-util/src/validate.rs#L17

The plan is to merge the `runtime-rs` branch into Kata's `main` branch soon.

## CoCo Plan

Once  the `runtime-rs` branch has been merged into Kata's `main` branch, we should make this repo consume the `kata-sys-util` crate as a dependency.

## Why bother?

The function is only ~18 lines of code, so is it worth doing this? I would say yes for the following reasons:

### Unit tests

The original version of the code comes with a set of tests, whereas the version of the PR in this repo does not.

### Maintenance issues

If there are multiple copies of the code, who's going to maintain them and ensure they stay in sync, with the latest fixes and improvements?

Rust eschews the golang approach of code copying (aka "vendoring") since [crates.io](https://crates.io/) makes that approach unnecessary.

### Security

This is the more important reason. The backstory for my raising https://github.com/kata-containers/kata-containers/pull/1521 was that container / sandbox IDs have a habit of ending up as part of path names for a container/sandbox specific data store. In CLI parlance...

```bash
$ cid="foo"
$ tree "/run/coco/containers/${cid}/"
/run/coco/containers/foo/
├── bar.json
└── baz.config
```

At some point, that path will be deleted:

```bash
$ sudo rm -rf "/run/foo/containers/${cid}/"
```

That _seems_ reasonable. How about now?

```bash
$ cid='../../../sbin/init'
$ sudo /bin/rm "/run/foo/containers/${cid}/"
```

Ouch. Not so good. This is just an example, but the point is that the function in question provides a basic level of protection against these sorts of abuses (see the unit tests on the origional PR for further details).

/cc @YangLiang3, @quanweiZhou, @bergwolf for thoughts.

In summary, we should try to reuse code as much as possible in CoCo, but avoid copying it if at all possible.

But if we _do_ have to copy code for some unusual situation, ensure that it is trivial for any interested parties to determine the original source location and authorship, aka it's provenance.

**Comment by jiazhang0** [comment]
@jodh-intel Agreed. In addition, I want to confirm whether this issue must be resolved before merging https://github.com/confidential-containers/enclave-cc/pull/14

**Comment by haosanzi** [comment]
Hi, the kata runtime-rs branch has been merged into Kata's [main branch](https://github.com/kata-containers/kata-containers/tree/main/src/runtime-rs).

**Comment by YangLiang3** [comment]
https://github.com/confidential-containers/enclave-cc/blob/b6d4d6cd59384a75f11f862fe5f6152b4b3c34de/src/enclave-agent/src/main.rs#L57
@jodh-intel  Could we please close this issue? Thanks for the great suggestion
